energy-integration/governance-v2/src/lib.rs

```diff
@@ -25,34 +27,76 @@ pub trait GovernanceV2:
     + views::ViewsModule
     + energy_query::EnergyQueryModule
 {
-    /// Used to deposit tokens for "payable" actions.
-    /// Funds will be returned if the proposal is defeated.
-    /// To keep the logic simple, all tokens have to be deposited at once
+    /// Used to deposit tokens to gather threshold min_fee.
+    /// Funds will be returned if the proposal is canceled.
     #[payable("*")]
     #[endpoint(depositTokensForProposal)]
     fn deposit_tokens_for_proposal(&self, proposal_id: ProposalId) {
         self.require_caller_not_self();
         self.require_valid_proposal_id(proposal_id);
 
-        let depositor_mapper = self.payments_depositor(proposal_id);
-        require!(depositor_mapper.is_empty(), "Payments already deposited");
-
-        let required_payments = self.required_payments_for_proposal(proposal_id).get();
         require!(
-            !required_payments.is_empty(),
-            "This proposal requires no payments"
+            !self.proposal_reached_min_fees(proposal_id),
+            MIN_FEES_REACHED
         );
 
-        let actual_payments = self.call_value().all_esdt_transfers();
+        let additional_fee = self.call_value().single_esdt();
         require!(
-            actual_payments == required_payments,
-            "Invalid payments, must match the required payments"
+            self.governance_token_id()
+                .get()
+                .eq(&additional_fee.token_identifier),
```

Comment:
Use `==` operator instead of manually calling `eq`

Reply:
Do no resolve conversations which are not yet resolved. You should resolve conversations only after commit is pushed to branch.